mars_street_view/tests/conftest.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+"""Configure fixtures for unit and functional tests."""
+import os
+import pytest
+# from webob import multidict
+from sqlalchemy import create_engine
+from pyramid import testing
+from mars_street_view.models import DBSession, Base
+from mars_street_view.scripts.initializedb import CAMERAS
+
+TEST_DATABASE_URL = 'sqlite:////tmp/test_db.sqlite'
+
+
+@pytest.fixture(scope='session')
+def global_environ(request):
+    """Establish test database url as a fixture for entire session."""
+    prior = os.environ.get('MARS_DATABASE_URL', '')
+    os.environ['MARS_DATABASE_URL'] = TEST_DATABASE_URL
+
+    def revert():
+        os.environ['MARS_DATABASE_URL'] = prior
+
+    request.addfinalizer(revert)
+    return True
+
+
+@pytest.fixture(scope='session')
+def config_uri():
+    """Establish configuration uri for initialization."""
+    parent_dir = os.path.dirname(__file__)
+    gparent_dir = os.path.dirname(parent_dir)
+    ggparent_dir = os.path.dirname(gparent_dir)
+    return os.path.join(ggparent_dir, 'testing.ini')
+
+
+@pytest.fixture(scope='session')
+def sample_data_uri():
+    """Establish configuration uri for initialization."""
+    return os.environ.get('SAMPLE_DATA_PATH')
+
+
+@pytest.fixture()
+def sqlengine(request):
+    """Return sql engine."""
+    engine = create_engine(TEST_DATABASE_URL)
+    DBSession.configure(bind=engine)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
```

Comment:
Here's the reason your tests take 20 minutes to run.  You are setting up and tearing down the entire database for each and every test function.  This fixture should be at session scope.